sql/core/src/test/scala/org/apache/spark/sql/execution/PlannerSuite.scala

```diff
@@ -223,6 +222,18 @@ class PlannerSuite extends SharedSQLContext {
     }
   }
 
+  test("collapse adjacent repartitions") {
```

Comment:
Decided to leave this here for now; this test isn't likely to be brittle and we don't really need other more complex tests for this feature, so it's fine to have a somewhat more end-to-end whole-optimizer test for this here.